lib/puma/request.rb

```diff
@@ -38,7 +38,7 @@ def handle_request(client, lines)
 
       env[PUMA_SOCKET] = io
 
-      if env[HTTPS_KEY] && io.peercert
+      if env[HTTPS_KEY] && io.methods.include?("peer_cert") && io.peercert
```

Comment:
Why is this necessary, and why are you checking for a different method than the one that you're actually calling?

Reply:
Its unnecessary, I will remove it.